app/logic/Mail/SQL/SQLEMail.ts

```diff
@@ -232,7 +232,7 @@ export class SQLEMail {
     email.pID = typeof(row.pID) == "number"
       ? sanitize.integer(row.pID, null)
       : sanitize.string(row.pID, null);
-    email.id = sanitize.nonemptystring(row.messageID);
+    email.id = sanitize.nonemptystring(row.messageID, "");
```

Comment:
Why?

Reply:
ActiveSync doesn't expose an email's message ID, so we'll only have the message ID if we actually download the message, but we still need to be able to save the message, otherwise we'll forget that it exists.